wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java

```diff
@@ -204,10 +219,140 @@ public void onPause() {
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         mappedIn = new MappedIn(getApplication());
+        setContentView(R.layout.activity_main);
+
+        Toolbar toolbar =  findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        final DrawerLayout drawer =  findViewById(R.id.drawer_layout);
+        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
+                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
+        drawer.setDrawerListener(toggle);
+        toggle.syncState();
+
         context = this;
         self = this;
-        setContentView(R.layout.activity_main);
+
         AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
+
+        LocationGenerator amenity = new LocationGenerator() {
```

Comment:
All this location generator stuff shouldn't be here. We should move it to the VenuesCallback.